### ISSUE TITLE
Testing with Session Variables Used in Application Controller

### ISSUE THREAD

**Issue by aximuseng** [rails/rails#39053]
I have a Rails 5 app with logic that sets a current tenant user role via a session variable.  The app works fine.  Looked to add some tests but they all fail at this session point.  I basically have logic like this:

@current_enrollment = Enrollment.Find(session[:enrollment_id])

session[:enrollment_id] is set on login.

As mentioned the actual app works fine.  I have this enrollment used in some of my Pundit authorization and it returns `nil` every time in the tests.  On a hunch to rule our Pundit I hard coded the enrollment in my Application controller then I had a second failure with my multi-tenant code.  I have my current tenant stored in session[:company_id].  This returns nil also.

I ended up going through all my code and hard coding these to match my fixture data and the tests run fine (at least with out the session related errors).

**Rails version**: Rails 5.2.4.2
**Ruby version**: 2.5.5

I did lots of digging - I suspect it my be related to the fact that I upgraded this app from Rails 4 to 5.  I recall back in those days I had a few tests working no problem. This may be related as some background: https://github.com/rails/rails/issues/25394.  This app also uses Devise.  I hope that this is just some small error in my configuration.

**Comment by jonathanhefner** [comment]
@aximuseng Could you post a minimal script that reproduces the issue?  Here is a [template](https://github.com/rails/rails/blob/master/guides/bug_report_templates/action_controller_gem.rb) that you can use.

**Comment by aximuseng** [comment]
Your reply triggered me to build a test app to try and reproduce this error.  In the process I had another dive into my code.  I found the culprit - still not sure why it fails in the test environment but the short version is I had a this set up to skip this setting of the enrollment if the tenant was not set.  This works in development / production but not in test.  I tweaked my code and the tests now run. 👍
